### PR TITLE
Login system back end

### PR DESCRIPTION
**Pull request by nwanderson009**
login back end is done

### CODE REVIEW

File: LoginBackEnd.mjs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import cors from "cors"
+import express from "express"
+import pkg from '../Team8/database.mjs';
+const { setUpDB, getCards, getUsers, getClient } = pkg;
```
Comment (brainwave0) [inline LoginBackEnd.mjs]: This looks overly complicated. Couldn't it be done in the following way?

```suggestion
import { setUpDB, getCards, getUsers, getClient } from '../database.mjs';
```

File: LoginBackEnd.mjs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+app.use(cors());
+
+app.post('/login', async (req, res) => {
+    console.log("login request recieved");
+    var result;
+    const { email, password } = req.body;
+    await setUpDB();
+    const users = getUsers();
+    if (await users.findOne({ email: email, password: password }) != null) { // if user can be found in database
+        result = { message: "login successful", user: await users.findOne({ email: email, password: password })}; // send success message and user
```
Comment (brainwave0) [inline LoginBackEnd.mjs]: Just to simplify, and also save time on database calls:

```suggestion
    const user = await users.findOne({ email: email, password: password });
    if (user != null) { // if user can be found in database
        result = { message: "login successful", user: user}; // send success message and user
```

File: LoginBackEnd.mjs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    var result;
+    const { email, password } = req.body;
+    await setUpDB();
+    const users = getUsers();
+    if (await users.findOne({ email: email }) != null) { //if there already exists a user with the given email
+        result = {message: "user already exists"};
+    }
+    else {
+        await users.insertOne({ email: email, password: password }); //add user to database
+        result = { message: "registration successful", user: users.findOne({ email: email, password: password })}
```
Comment (brainwave0) [inline LoginBackEnd.mjs]: You could maybe just do this:

```suggestion
        const user = { email: email, password: password };
        await users.insertOne(user); //add user to database
        result = { message: "registration successful", user: user}
```

File: database.mjs [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { createRequire } from 'module'
+const require = createRequire(import.meta.url);
+const { MongoClient } = require('mongodb');
```
Comment (brainwave0) [inline database.mjs:3]: What is this `createRequire` for?

File: database.mjs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    return cards;
+}
+function getUsers() {
+    return users;
+}
+function getClient() {
+    return client;
+}
+
+export default{ setUpDB, getDB, getCards, getUsers, getClient };
```
Comment (brainwave0) [inline database.mjs:32]: You might want to remove the `default`. I'm not sure, but I think this exports the functions as part of a single object, making you have to deconstruct it when you import.

**Review by brainwave0 — COMMENTED** [review]
Looks good. Unless anyone else wants to review, we should merge it.

File: database.mjs (hunk 4 above)
Comment (nwanderson009) [reply database.mjs:3]: this allows for the require syntax to be used in an mjs file, where it is otherwise undefined.

**Pull request merged**